main/commons/src/main/java/org/cryptomator/common/vaults/VaultListManager.java

```diff
@@ -124,9 +125,9 @@ private Vault create(VaultSettings vaultSettings) {
 	}
 
 	private static VaultState.Value determineVaultState(Path pathToVault) throws IOException {
-		if (!CryptoFileSystemProvider.containsVault(pathToVault, MASTERKEY_FILENAME)) {
+		if (!CryptoFileSystemProvider.containsVault(pathToVault, VAULTCONFIG_FILENAME, MASTERKEY_FILENAME)) {
 			return VaultState.Value.MISSING;
-		} else if (Migrators.get().needsMigration(pathToVault, MASTERKEY_FILENAME)) {
+		} else if (Migrators.get().needsMigration(pathToVault, VAULTCONFIG_FILENAME, MASTERKEY_FILENAME)) {
```

Comment:
If the vault config file of a _version 8_ vault is missing, but the `d` directory and a masterkey file present, it is assumed that the vault "exists" and does not need migration.

Reply:
Depends on https://github.com/cryptomator/cryptofs/issues/103

Reply:
fixed in 76d1875